dirsrvtests/tests/suites/schema/schema_replication_origin_test.py

```diff
@@ -0,0 +1,237 @@
+# --- BEGIN COPYRIGHT BLOCK ---
+# Copyright (C) 2023 Red Hat, Inc.
+# All rights reserved.
+#
+# License: GPL (version 3 or any later version).
+# See LICENSE for details.
+# --- END COPYRIGHT BLOCK ---
+#
+import json
+import logging
+import re
+import time
+import ldap
+import pytest
+from lib389._constants import SUFFIX, ReplicaRole, DEFAULT_SUFFIX
+from lib389.topologies import create_topology
+from lib389.replica import Agreements
+from lib389.schema import Schema
+from lib389.idm.user import UserAccounts
+from lib389.cli_base import LogCapture
+from lib389.utils import *
+
+logging.getLogger(__name__).setLevel(logging.DEBUG)
+log = logging.getLogger(__name__)
+
+def pattern_errorlog(file, log_pattern):
+    """Check for a pattern in the error log file."""
+
+    try:
+        pattern_errorlog.last_pos += 1
+    except AttributeError:
+        pattern_errorlog.last_pos = 0
+
+    found = None
+    log.debug("_pattern_errorlog: start at offset %d" % pattern_errorlog.last_pos)
+    file.seek(pattern_errorlog.last_pos)
+
+    # Use a while true iteration because 'for line in file: hit a
+    # python bug that break file.tell()
+    while True:
+        line = file.readline()
+        log.debug("_pattern_errorlog: [%d] %s" % (file.tell(), line))
+        found = log_pattern.search(line)
+        if ((line == '') or (found)):
+            break
+
+    log.debug("_pattern_errorlog: end at offset %d" % file.tell())
+    pattern_errorlog.last_pos = file.tell()
+    return found
+
+
+def trigger_update(topology, user_rdn, num):
+    """It triggers an update on the supplier. This will start a replication
+    session and a schema push
+    """
+
+    users_s = UserAccounts(topology.ms["supplier1"], DEFAULT_SUFFIX)
+    user = users_s.get(user_rdn)
+    user.replace('telephonenumber', str(num))
+
+    # wait 10 seconds that the update is replicated
```

Comment:
I would change the comment to: wait until the update is replicated (until up to x seconds)
IMHO using  "for loop in range(30):"   is more pythonic
And I would rather wait up to 30s because we had to increase the replication timeout in a lot of testcases.
 10s is very often working fine but from time to time the test vm is really crawling and we get timeout while nothing is broken ...